identity-iota/src/did/doc/document.rs

```diff
@@ -381,6 +382,19 @@ impl Document {
 
     Ok(DID::encode_key(message_id.encode_hex().as_bytes()))
   }
+
+  pub fn insert_service(&mut self, service: Service) -> bool {
+    if service.id().fragment().is_none() {
+      false
+    } else {
+      self.document.service_mut().append(service.serde_into().unwrap())
```

Comment:
`service.serde_into().unwrap()`

This will actually cause additional properties to be dropped - also errors should be handled or a comment about why `unwrap` is "safe".

but you can actually remove `.serde_into().unwrap()` completely if you change this

```rust
type BaseDocument = CoreDocument<Properties, Object, ()>;
pub type Signer<'a, 'b, 'c> = DocumentSigner<'a, 'b, 'c, Properties, Object, ()>;
pub type Verifier<'a> = DocumentVerifier<'a, Properties, Object, ()>;
```

to this

```rust
type BaseDocument = CoreDocument<Properties, Object, Object>;
pub type Signer<'a, 'b, 'c> = DocumentSigner<'a, 'b, 'c, Properties, Object, Object>;
pub type Verifier<'a> = DocumentVerifier<'a, Properties, Object, Object>;
```

The third generic type in `CoreDocument` is used by `Service` so it just needs to change from unit `()` to `Object`